### DISCUSSION TITLE
model confound condition in single-trial estimates

### DISCUSSION THREAD

**Discussion by assuntaciarlo** [cvnlab/GLMsingle#146] [Q&A] [answered]
Hello,

I'm dealing with the following block design (x 18 trials within each run)

![Screenshot 2024-07-09 193119](https://github.com/cvnlab/GLMsingle/assets/79250141/f3cf879e-2ee5-4f19-aca1-79239a8a4f80)

and would like to compute single trial estimates for the second condition only (the yellow one), while modelling the first one as a confound. Which would be the best way to deal with this design using GLMSingle? The condition durations are 5 and 15 sec, respectively.

Many thanks in advance!

**Comment by kendrickkay** [comment]
If you don't care about the response to the first one at all (e.g., don't even want the beta weight associated with that one), then you could use the extraregressors input argument to specify how exactly you want to model that variance.  For example, you could include a single regressor covering all instances/trials of the first one, or maybe you want to include a separate regressor for each trial associated with the first one.   Then, you could specify the second condition (15 s) as the experimental condition of interest, and use a stimdur of 15.

**Reply by assuntaciarlo** [reply]
Many thanks for the quick reply! Ok, then in this case I should use the canonical HRF to create the regressor for the first condition like I did before and add it as extraregressor, whereas the HRF for the main condition will be chosen by your optimisation. Do you think that it is a reasonable solution to explain the variance of the first predictor? Otherwise I guess I would need additional data to know a priori which HRF to use for the first predictor.

**Reply by kendrickkay** [reply, marked as answer]
Yeah, the easiest thing is to do what you said (just assume the canonical HRF, e.g. getcanonicalhrf.{m,py}).  I think it's probably good enough...     

(Alternatively, you'd have to do something like, first run GLMsingle to try to estimate the HRF for each voxel, and then use that to generate the extragressor, and then run it a second time...     But this seems like a lot of work for unknown benefits.  I guess, you can see how things look...)

**Reply by assuntaciarlo** [reply]
Yes, I guess the first option should be fine enough. Many thanks for the feedback!
